src/condor_submit.V6/factory_test.cpp

```diff
@@ -1038,27 +503,39 @@ main( int argc, const char *argv[] )
 			if (verbose) { fprintf(stdout, "# no more jobs to materialize\n"); }
 			break;
 		}
-		if (DashDryRun) fprintf(out, ".\n");
+		if (out) fputs("\n", out);
 	}
-	if (DashDryRun) fprintf(out, "%d job(s) dry-run to cluster %d.\n", num_jobs, cluster_id);
 
+	fprintf(stdout, "%d job(s) materialized to cluster %d.\n", num_jobs, cluster_id);
+
+	NewProcOutFp = nullptr;
 	if (free_out && out) {
 		fclose(out); out = NULL;
 	}
 
-	delete factory;
+	// this also deletes the factory object
+	delete ClusterObj;
 	return 0;
 }
 
-void usage()
+void usage(FILE* out)
 {
-	fprintf(stderr, "Usage: %s [options] -clusterad <classad-file> - | <digest-file>\n", MyName);
-	fprintf(stderr, "\t-clusterad <classad-file>\tRead The ClusterAd <classad-file>\n");
-	fprintf(stderr, "    [options] are\n");
-	fprintf(stderr, "\t-digest <digest-file>\tRead Submit commands from <digest-file>\n");
-	fprintf(stderr, "\t-items <items-file>\tRead Submit itemdata from <items-file>\n");
```

Comment:
I think the only thing missing at this point is to restore the -items line for the factory_test.cpp usage()